File: Modules/ServerTypes/CustomerManagement/CustomerContactForm.ts
import { initFormType, PrefixedContext, StringEditor, TextAreaEditor } from "@serenity-is/corelib";

export interface CustomerContactForm {
    ContactName: StringEditor;
    Title: StringEditor;
    Phone: StringEditor;
    Email: StringEditor;
    Notes: TextAreaEditor;
}

export class CustomerContactForm extends PrefixedContext {
    static readonly formKey = 'CustomerManagement.CustomerContact';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CustomerContactForm.init) {
            CustomerContactForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;

            initFormType(CustomerContactForm, [
                'ContactName', w0,
                'Title', w0,
                'Phone', w0,
                'Email', w0,
                'Notes', w1
            ]);
        }
    }
}

File: Modules/CustomerManagement/CustomerContact/CustomerContactForm.cs
namespace MapPin.CustomerManagement.Forms;

[FormScript("CustomerManagement.CustomerContact")]
[BasedOnRow(typeof(CustomerContactRow), CheckNames = true)]
public class CustomerContactForm
{
    public string ContactName { get; set; }
    public string Title       { get; set; }
    public string Phone       { get; set; }
    public string Email       { get; set; }
    public string Notes       { get; set; }
}
